cirq/ops/common_channels.py

```diff
@@ -187,3 +189,603 @@ def depolarize(p: float) -> DepolarizingChannel:
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'PF({!r})'.format(self._p)
+
+
+def phase_flip(p: float) -> PhaseFlipChannel:
+    """
+    Returns a PhaseFlipChannel that flips a qubit's phase with probability p
+
+    This channel evolves a density matrix via
+           \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+    With
+        M_0 = \sqrt{p} \begin{bmatrix}
+                            1 & 0  \\
+                            0 & 1
+                       \end{bmatrix}
+        M_1 = \sqrt{1-p} \begin{bmatrix}
+                            1 & 0 \\
+                            0 & -1
+                         \end{bmatrix}
+
+    Args:
+        p: the probability of a phase flip
+
+    Raises:
+        ValueError if p is not a valid probability.
+    """
+    return PhaseFlipChannel(p)
+
+
+class BitFlipChannel(raw_types.Gate):
+    """
+    Flips a qubit from 1 to 0 state or vice versa according to some
+    probability.
+    """
+
+    def __init__(self, p) -> None:
+
+        """
+        Construct a channel that flips a qubit with probability p
+
+        This channel evolves a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+        With
+            M_0 = \sqrt{p} \begin{bmatrix}
+                                1 & 0  \\
+                                0 & 1
+                           \end{bmatrix}
+            M_1 = \sqrt{1-p} \begin{bmatrix}
+                                0 & 1 \\
+                                1 & -0
+                             \end{bmatrix}
+
+        Args:
+            p: the probability of a bit flip
+
+        Raises:
+            ValueError if p is not a valid probability.
+        """
+
+        def validate_probability(p, p_str):
+            if p < 0:
+                raise ValueError('{} was less than 0.'.format(p_str))
+            elif p > 1:
+                raise ValueError('{} was greater than 1.'.format(p_str))
+            return p
+
+        self._p = validate_probability(p, 'p')
+        self._delegate = AsymmetricDepolarizingChannel(1. - p, 0., 0.)
+
+    def _channel_(self) -> Iterable[np.ndarray]:
+        # Return just the I and X pieces.
+        return list(self._delegate._channel_())[:2]
+
+    def _eq_tuple(self):
+        return (BitFlipChannel, self._p)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.bit_flip(p={!r})'.format(self._p)
+
+    def __str__(self) -> str:
+        return 'bit_flip(p={!r})'.format(self._p)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'BF({!r})'.format(self._p)
+
+
+def bit_flip(p: float) -> BitFlipChannel:
```

Comment:
Have `p` default to `None`, change return type to `Union[ops.XPowGate, BitFlipChannel]`, declare an overload with no argument returning `ops.XPowGate` and an overload with the non-optional argument returning `BitFlipChannel`. If `p is None` then return `cirq.X`.

Reply:
For this issue and the one above I wasn't exactly sure how you wanted things overloaded, so is how I've done it somewhat close to what you are after ?